lib/git_service/commit.rb

```diff
@@ -23,21 +23,63 @@ def rugged_diff(other_ref = parent_oids.first)
       other_commit.diff(rugged_commit)
     end
 
+    def formatted_author
+      "#{rugged_commit.author[:name]} <#{rugged_commit.author[:email]}>"
+    end
+
+    def formatted_author_date
+      rugged_commit.author[:time].to_time.strftime("%c %z")
+    end
+
+    def formatted_committer
+      "#{rugged_commit.committer[:name]} <#{rugged_commit.committer[:email]}>"
+    end
+
+    def formatted_committer_date
+      rugged_commit.committer[:time].to_time.strftime("%c %z")
+    end
+
+    def formatted_commit_stats
+      diff.file_status.map do |file, stats|
+        file_stats  = file
```

Comment:
Note that the following line modify the file object directly...we may want this line to be `files_stats = file.dup`

Reply:
Hmmm, then I guess this was happening previously as well.  I will look into it.

Reply:
Fixed now.